DEVELOP.md

```diff
@@ -0,0 +1,294 @@
+  code samples in documentation.
+* [pytest-cov](https://pypi.python.org/pypi/pytest-cov/1.8.1) to generate
+  test-coverage reports. Since our tests intend to be comprehensive,
+  test-coverage of less than 100% is not acceptable. Note that code 100%
+  coverage does not indicate comprehensive tests.
+* [flake8](https://pypi.python.org/pypi/flake8) to enforce a [consistent code
+  style](https://www.python.org/dev/peps/pep-0008/).
+
+The test suite requires a number of additional Python packages to run which can
+be installed using::
+
+    pip install -r requirements-test.txt
+
+### Running tests
+
+The test suite can be run against the source code and documentation using the
+following command:
+
+    py.test rig docs
+
+This runs a subset of the full test suite which does not require an attached
+SpiNNaker board. The test suite automatically includes all doctest blocks in
+docstrings as well as doctest blocks in documentation files ending in
+`_doctest.rst`.
+
+#### Testing against local hardware
+
+Some tests require a connected SpiNNaker system.
+
+**For a minimal live-hardware test,** the attached SpiNNaker system must not be
+booted and have at least 2x2 SpiNNaker chips with the majority of cores
+operational. To include tests which can operate against a generic SpiNNaker
+system, run:
+
+    py.test rig docs --spinnaker SPINN_HOSTNAME WIDTH HEIGHT
+
+**For a comprehensive live-hardware test,** the test suite should be run
+against a system consisting of a single SpiNN-5 board like so:
+
+    py.test rig docs \
+            --spinnaker SPINN_HOSTNAME 8 8 --spinn5
+            --bmp BMP_HOSTNAME
+
+### Test suite command line option reference
+
+The test suite supports the following commandline arguments:
+
+* `--spinnaker SPINN_HOSTNAME WIDTH HEIGHT` -- run tests against the specified
+  SpiNNaker system.
+* `--spinn5` -- include tests which specifically require a SpiNN-5 board to be
+  attached.
+* `--bmp BMP_HOSTNAME` -- run BMP communication tests against the suggested
+  BMP. This option is only recommended for use against a single SpiNN-5 board
+  since (by default) it power-cycles the board in 'slot 0'.
+* `--no-boot` -- Skip the power-cycle BMP test and the SpiNNaker booting test.
+  When this option is used, the attached SpiNNaker hardware must be already
+  booted.
+
+#### Testing against remote hardware
+
+Booting a SpiNNaker board requires the (reliable) sending of packets to UDP
+port 54321 which is frequently blocked by ISPs and is not reliable (since UDP
+gives no guarantees, especially on the open internet). As a result, a proxy
+server must be used to communicate with the board. A utility such as
+[`spinnaker_proxy`](https://github.com/project-rig/spinnaker_proxy) can be used
+alongside the test suite for this purpose.
+
+### Test coverage checking
+
+To get a test coverage report run one of the following:
+
+    # Summary printed on the commandline
+    py.test rig docs --cov rig
```

Comment:
Any strong feelings there? I dropped it since finding it very hard to use when branch testing is enabled.